main/src/main/scala/sbt/internal/FileManagement.scala

```diff
@@ -55,11 +53,8 @@ private[sbt] object FileManagement {
           watchLogger
         )
     } else {
-      if (Util.isWindows) new PollingFileRepository(FileAttributes.default)
-      else {
-        val service = Watched.createWatchService(pollInterval)
-        FileTreeRepository.legacy(FileAttributes.default _, (_: Any) => {}, service)
-      }
+      val service = Watched.createWatchService(pollInterval)
```

Comment:
Is this undoing Windows specific handling added in https://github.com/sbt/sbt/commit/e3625ca8b0781288f379eff6d6e6534998d64200, or it's no longer needed?

Reply:
It's no longer needed.

Reply:
Actually, that is mistaken. It is not needed using new code that isn't part of this PR so I need to restore the windows specific code until I open that PR. It works fine on my VM, but CI fails with regular file repository. I'm 90% sure that CI will pass with some changes from https://github.com/sbt/io/pull/226.